libblkid/src/superblocks/drbd.c

```diff
@@ -109,7 +109,7 @@ struct meta_data_on_disk_9 {
 	struct peer_dev_md_on_disk_9 peers[DRBD_PEERS_MAX];
 	uint64_t history_uuids[HISTORY_UUIDS];
 
-	char padding[0] __attribute__((aligned(4096)));
+	uint8_t padding[2704];
```

Comment:
Maybe all we need is to remove the padding from the struct at all and add 

   #define DRBD_METADATA_OFFSET 4096

and use it for blkid_probe_get_buffer() rather than depend on sizeof() when it counts the offset of the struct.